dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/Clients/GeminiChatCompletionClient.cs

```diff
@@ -0,0 +1,611 @@
+    /// <param name="apiKey">Api key for GoogleAI endpoint</param>
+    /// <param name="logger">Logger instance used for logging (optional)</param>
+    public GeminiChatCompletionClient(
+        HttpClient httpClient,
+        string modelId,
+        string apiKey,
+        ILogger? logger = null)
+        : base(
+            httpClient: httpClient,
+            logger: logger)
+    {
+        Verify.NotNullOrWhiteSpace(modelId);
+        Verify.NotNullOrWhiteSpace(apiKey);
+
+        this._modelId = modelId;
+        this._streamJsonParser = new StreamJsonParser();
+        this._chatGenerationEndpoint = new Uri($"https://generativelanguage.googleapis.com/v1beta/models/{this._modelId}:generateContent?key={apiKey}");
+        this._chatStreamingEndpoint = new Uri($"https://generativelanguage.googleapis.com/v1beta/models/{this._modelId}:streamGenerateContent?key={apiKey}&alt=sse");
+    }
+
+    /// <summary>
+    /// Represents a client for interacting with the chat completion gemini model via VertexAI.
+    /// </summary>
+    /// <param name="httpClient">HttpClient instance used to send HTTP requests</param>
+    /// <param name="modelId">Id of the model supporting chat completion</param>
+    /// <param name="bearerKeyProvider">Bearer key provider used for authentication</param>
+    /// <param name="location">The region to process the request</param>
+    /// <param name="projectId">Project ID from google cloud</param>
+    /// <param name="logger">Logger instance used for logging (optional)</param>
+    public GeminiChatCompletionClient(
+        HttpClient httpClient,
+        string modelId,
+        Func<string> bearerKeyProvider,
+        string location,
+        string projectId,
+        ILogger? logger = null)
+        : base(
+            httpClient: httpClient,
+            logger: logger,
+            bearerKeyProvider: bearerKeyProvider)
+    {
+        Verify.NotNullOrWhiteSpace(modelId);
+        Verify.NotNullOrWhiteSpace(location);
+        Verify.NotNullOrWhiteSpace(projectId);
+
+        this._modelId = modelId;
+        this._streamJsonParser = new StreamJsonParser();
+        this._chatGenerationEndpoint = new Uri($"https://{location}-aiplatform.googleapis.com/v1/projects/{projectId}/locations/{location}/publishers/google/models/{this._modelId}:generateContent");
+        this._chatStreamingEndpoint = new Uri($"https://{location}-aiplatform.googleapis.com/v1/projects/{projectId}/locations/{location}/publishers/google/models/{this._modelId}:streamGenerateContent?alt=sse");
+    }
+
+    /// <inheritdoc/>
+    public async Task<IReadOnlyList<ChatMessageContent>> GenerateChatMessageAsync(
+        ChatHistory chatHistory,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        var state = ValidateInputAndCreateChatCompletionState(chatHistory, kernel, executionSettings);
```

Comment:
Does this have the same issue addressed in https://github.com/microsoft/semantic-kernel/pull/5376?